pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/RealtimeSegmentStatsHistory.java

```diff
@@ -233,9 +235,21 @@ public synchronized boolean isEmpty() {
     return getNumntriesToScan() == 0;
   }
 
+  public void setMinIntervalBetweenUpdatesMillis(long millis) {
+    _minIntervalBetweenUpdatesMillis = millis;
+  }
+
   public synchronized void addSegmentStats(SegmentStats segmentStats) {
```

Comment:
Many methods in this class are 'synchronized' - I am not sure if all private members need careful access; the newly added setMinIntervalBetweenUpdatesMillis is updating a long value without a lock. Is that safe? 

It might be good if we have some class level comments on thread-safety related aspects of the class so future modifiers have the right context.

Reply:
Since the new method is public, there is no guarantee that other parts of the code won't call it any other time. Should the interval be really decided outside this class? If we move it inside this class, then we risk fewer issues with setters getting the way. This is mainly from a pure correctness point of view.